### DISCUSSION TITLE
[Proposal] The `not` keyword as an alternative syntax for the logical not operator (!)

### DISCUSSION THREAD

**Discussion by luuhq** [dotnet/csharplang#1013] [General]
Not sure if this has been brought up before, would love your thoughts on it.

### Background
Many of my colleagues and I personally prefer using `== false` in a negative comparison instead of the bang sign `!`. The reason being, it's easier to spot, especially when one is debugging a live-site/SEV in the middle of the night. For example: `if (!isUserAuthorized)` vs `if (isUserAuthorized == false)`. 

### Proposal
The proposal here is to introduce a `not` keyword as an alternative syntax for the unary logical not operator (`!`). This keyword should not require a set of parentheses after it (just like the existing `!` operator). For example: `if (not isUserAuthorized)`, which is exactly equivalent to `if (!isUserAuthorized)`.

This syntax should also work: `if (not(isUserAuthorized))`.

For what it's worth, C++ also has this `not` keyword (among `not_eq`, `and,` and `or`).

**Comment by tannergooding** [comment]
There are a few existing issues. The oldest/most prevalent is probably: https://github.com/dotnet/csharplang/issues/27

**Comment by andrevdl** [comment]
when would uses as:
> if (not(isUserAuthorized))

You could think that "not" is method defined in the same, derived or static using class.
Also can it break existed code, when somebody defined a method that is be named "not".

**Comment by AustinBryan** [comment]
> For what it's worth, C++ also has this

Congrats. C++ has a lot of problems that C# doesn't have. This is a very weak argument. If you really need it just create a `public static not(bool b) => !b;`, but it in a static class, use a `using static` on every file (something you could configure VS to do for you automatically, and use it like `if(not (b));`.

No need for a keyword that's probably going to be in use for more pattern matching stuff when you can easily do it yourself.
